pkg/deploy/job.go

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright (c) 2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package deploy
+
+import (
+	"context"
+
+	orgv1 "github.com/eclipse/che-operator/pkg/apis/org/v1"
+	"github.com/eclipse/che-operator/pkg/util"
+	"github.com/sirupsen/logrus"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// SyncJobToCluster deploys new instance of given job
+func SyncJobToCluster(instance *orgv1.CheCluster, job *batchv1.Job, clusterAPI ClusterAPI) error {
```

Comment:
pls. have a look at pvc.go for instance.
The implementation should be consistent

Reply:
What do you think is not consistent?

Reply:
It must be done like this
https://github.com/eclipse/che-operator/blob/master/pkg/deploy/pvc.go

Reply:
@tolusha which specific part do you think is bad? Return type?

Reply:
`job.go` should contain 3 methods:
`getClusterJob` - to return existed job by name
`getSpecJob` - to return a spec for a new job (+ set owner)
`syncJobToCluster` - to compare spec and cluster jobs, print differences, update the job if necessary and return the status.

It is how we did for others objects.

Reply:
@tolusha jobs are running after creation. It has a little sense to update one. Are you sure we need to do diff and update?

Reply:
I'll try to code it like ingress is done